app/services/case_updater_service.rb

```diff
@@ -13,7 +13,13 @@ def call
     ActiveRecord::Base.transaction do
       begin
         @kase.assign_attributes(@params)
-        if @kase.changed?
+        # properties is JSON object, if one of keys was not included 
+        # but later this key with value of nil is added, 
+        # it will be treated as changed which will give a misleading message 
+        # as from user's point of view, he/she doesn't change anything
+        # Each key within this JSON object will be tracked individually, 
+        # no need for tracking properties as the whole.
+        if (@kase.changed_attributes.keys - ["properties"]).present?
```

Comment:
What is this line doing? If there are keys other than 'properties' would this not return a false positive? e.g.

```
irb(main):001:0> a = ['a', 'b', 'c']
irb(main):002:0> a - ['c']
=> ["a", "b"] # this is still .present?


# whereas this wouldn't be:
irb(main):005:0> a = ['a']
irb(main):006:0> a - ['a']
=> []
```

Is this okay? :)

Reply:
@willmcb , yes that is my intention :),   so if c is the 'properties, for the following cases, 1, 2 will be considered as changed 
1 - ['a', 'b', 'c'] - ['c'] 
2 - ['a', 'b'] - ['c']
3 - []-['c']
4 - ['c'] - ['c']

Reply:
Great thanks I thought it may have been but I thought I'd check :)